src/main/kotlin/the/coding/force/exploring_kotlin_coroutines/configuration/CoroutineConfig.kt

```diff
@@ -0,0 +1,14 @@
+package the.coding.force.exploring_kotlin_coroutines.configuration
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.reactor.asCoroutineDispatcher
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import reactor.core.scheduler.Schedulers
+
+@Configuration
+class CoroutineConfig {
+
+    @Bean
+    fun ioDispatcher(): CoroutineDispatcher = Schedulers.boundedElastic().asCoroutineDispatcher()
```

Comment:
Pode usar o Dispatcher de forma nativa do Kotlin

```kotlin 
import kotlinx.coroutines.CoroutineDispatcher
import kotlinx.coroutines.Dispatchers
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class CoroutineConfig {

    @Bean
    fun ioDispatcher(): CoroutineDispatcher = Dispatchers.IO

}

Reply:
faz sentido ser um bean? é que é algo estático e tal
e o caso de uso de cada dispatcher é diferente tb

Reply:
é necessário ser um bean? é que é algo estático e tal
e o caso de uso de cada dispatcher é diferente tb, não acho que precisa ser generico assim

Reply:
> é necessário ser um bean? é que é algo estático e tal e o caso de uso de cada dispatcher é diferente tb, não sei se precisa ser generico assim

Então mano lêdo, se formos usar o Dispatcher padrão do kotlin, não é preciso usar ele em forma de bean, já que eu consigo acessar esse dispatcher IO de forma estática, sem precisar que o spring gerencie as instâncias dessa classe em específico.

com isso eu poderia até mesmo na classe de serviço, usar o Dispatcher desse jeito 
```kotlin
  suspend fun delete(dataId: Long) {
        withContext(Dispatcher.IO) {
         }
    }
```

Reply:
> Pode usar o Dispatcher de forma nativa do Kotlin

É possível, pelo que entendi do assunto, usar o Dispatcher padrão já é uma solução boa geralmente, a menos que seja em um cenário mais específico.

O dispatcher padrão do Kotlin(pelo que entendi) não tem um gerenciamento preciso das threads, isso significa que conforme a carga da aplicação for crescendo, o número de threads pode crescer de forma indefinida, fazendo o uso de memória se tornar excessivo.
Com essa implementação aqui:
```kotlin
Schedulers.boundedElastic().asCoroutineDispatcher()
```
O `boundedElastic()` cria um pool de threads limitado especializado em operações de entrada e saída de dados, isso significa que caso a carga da aplicação cresça demais, o número de threads não vai subir de forma descontrolada

O `asCoroutineDispatcher()` faz com que o Kotlin gerencie esse pool de threads delimitado com as coroutines.

Mas na maioria dos casos, a implementação padrão do kotlin pode resolver, a menos que precisemos de um cenário com um estresse muito grande de IO

Reply:
commit: [improving logs, change Dispatcher strategy, remove webFlux dependency](https://github.com/the-coding-force/exploring-kotlin-coroutines__backend__spring-kotlin__poc/pull/13/commits/bcb1229dfa29acc3455c10a16adc4b18b4edd81d)